src/helm/benchmark/augmentations/cleva_perturbation.py

```diff
@@ -0,0 +1,722 @@
+        prob: float = 0.0
+        rare_word_prob: float = 0.05
+        consider_tone: bool = False
+        word_level_perturb: bool = True
+
+    name: str = "butter_finger"
+
+    # For downloading resources
+    ASSET_URL = "https://drive.google.com/uc?id=1p5mldLpKxI-63H8YEruGJghtD1dZJI8k"
+
+    def __init__(
+        self,
+        prob: float,
+        rare_char_prob: float = 0.05,
+        consider_tone: bool = False,
+        word_level_perturb: bool = True,
+    ):
+        # Assign parameters to instance variables
+        self.prob: float = prob
+        self.rare_char_prob: float = rare_char_prob  # How likely we will use rare Chinese characters
+        self.consider_tone: bool = (
+            consider_tone  # Should we take the tone of Pinyin into account when considering similar char/words
+        )
+        self.word_level_perturb: bool = word_level_perturb  # Whether we perturb text on the character or word level
+
+        # Ensure all necessary data are downloaded
+        output_dir = os.path.join("benchmark_output", "perturbations", self.name)
+        ensure_directory_exists(os.path.dirname(output_dir))
+        ensure_file_downloaded(source_url=self.ASSET_URL, target_path=output_dir, unpack=True, unpack_type="unzip")
+
+        # Load the data for the perturbation
+        with open(
+            os.path.join(
+                output_dir,
+                "pinyin_to_char.json" if self.consider_tone else "toneless_pinyin_to_char.json",
+            )
+        ) as f:
+            self.chinese_character_database: Dict[str, List[str]] = json.load(f)
+        with open(
+            os.path.join(
+                output_dir,
+                "pinyin_to_common_char.json" if self.consider_tone else "toneless_pinyin_to_common_char.json",
+            )
+        ) as f:
+            self.common_chinese_character_database: Dict[str, List[str]] = json.load(f)
+        with open(
+            os.path.join(
+                output_dir,
+                "pinyin_to_word.json" if self.consider_tone else "toneless_pinyin_to_word.json",
+            )
+        ) as f:
+            self.chinese_words_database: Dict[str, List[str]] = json.load(f)
+
+    @property
+    def description(self) -> PerturbationDescription:
+        return ButterFingerPerturbation.Description(name=self.name, robustness=True, prob=self.prob)
+
+    def perturb(self, text: str, rng: Random) -> str:
+        butter_text: str = ""
+        output: List[str] = jieba.lcut(text)
+        if self.word_level_perturb:
+            words_and_similar_word_dict_list = self.get_words_with_similar_pinyin(
+                output,
+                self.rare_char_prob,
+                self.chinese_character_database,
+                self.common_chinese_character_database,
+                self.chinese_words_database,
+                self.consider_tone,
+                rng,
+            )
+            for dict in words_and_similar_word_dict_list:
```

Comment:
Don't name a variable `dict` because that is a name of a Python built-in.

If you follow my later suggestion in `get_words_with_similar_pinyin`, then this just becomes

```
for original_word, similar_pinyin_words in words_and_similar_word_dict.items():
```